### PR TITLE
Remove deprecated build.py script

### PR DESCRIPTION
**Pull request by rachel-fenichel**
## The basics

- [x] I branched from develop
- [x] My pull request is against develop
- [x] My code follows the [style guide](https://developers.google.com/blockly/guides/modify/web/style-guide)

### Resolves

No filed issue.

### Proposed Changes

Remove the deprecated `build.py` script, which has been replaced by `npm run build` for a while now.

### Reason for Changes

Deprecated, and it's confusing that it was still around.

### Test Coverage

Build should still work (tested by CI)

### Documentation

Build.py is mentioned in two pages on devsite in the guides section, and once in the generated reference section.

The reference section is based on comments in code and will be updated when docs are regenerated.

I have a CL out for removing the mentions in the guides section, assigned to @moniika 

### Additional Information

Should I hold off on merging until after release week?

### CODE REVIEW

File: core/blockly.js [hunk 1]
```diff
@@ -61,10 +61,10 @@ goog.requireType('Blockly.Workspace');
 
 /**
  * Blockly core version.
- * This constant is overridden by the build script (build.py) to the value of the version
- * in package.json. This is done during the gen_core build step.
- * For local builds, you can pass --define='Blockly.VERSION=X.Y.Z' to the compiler
- * to override this constant.
+ * This constant is overridden by the build script (npm run build) to the value
+ * of the version in package.json. This is done during the gen_core build step.
```
Comment (cpcallen) [inline core/blockly.js:65]: Perhaps: "This is done during by the Closure Compiler during the buildCompressed gulp task."?

File: msg/messages.js [hunk 2]
```diff
@@ -8,8 +8,8 @@
  * @fileoverview English strings.
  * @author fraser@google.com (Neil Fraser)
  *
- * After modifying this file, either run "build.py" from the parent directory,
- * or run (from this directory):
+ * After modifying this file, either run "npm run build:langfiles" from the
+ * parent directory, or run (from this directory):
```
Comment (cpcallen) [inline msg/messages.js:12]: I'm about to patch this same comment in another PR.

**Review by cpcallen — APPROVED** [review]
You beat me to it.

One suggested comment improvement.

File: msg/messages.js (hunk 2 above)
Comment (rachel-fenichel) [reply msg/messages.js:12]: If you were going to hit these lines in your PR on build stuff, I can just close this and wait.

**Pull request merged**